packages/app-builder-lib/src/util/appFileCopier.ts

```diff
@@ -188,32 +188,40 @@ export async function computeNodeModuleFileSets(platformPackager: PlatformPackag
   const result = new Array<ResolvedFileSet>()
   let index = 0
   const NODE_MODULES = "node_modules"
-  const getRealSource = (source: string) => {
-    const parentDir = path.dirname(source)
+  const getRealSource = (name: string, source: string) => {
+    const normalizedPath = source.split(path.sep).join("/")
+
+    if (!normalizedPath.endsWith(name)) {
+      throw new Error("Path does not end with the package name")
+    }
+    const parentDir = path.normalize(normalizedPath.slice(0, -name.length - 1))
+
     // for the local node modules which is not in node modules
     if (!parentDir.endsWith(path.sep + NODE_MODULES)) {
       return parentDir
     }
     // use main matcher patterns, so, user can exclude some files !node_modules/xxxx
     return path.dirname(parentDir)
   }
-  const collectNodeModules = async (dep: NodeModuleInfo, destination: string) => {
+
+  const collectNodeModules = async (dep: NodeModuleInfo, realSource: string, destination: string) => {
     const source = dep.dir
-    const matcher = new FileMatcher(getRealSource(source), destination, mainMatcher.macroExpander, mainMatcher.patterns)
+    const matcher = new FileMatcher(realSource, destination, mainMatcher.macroExpander, mainMatcher.patterns)
     const copier = new NodeModuleCopyHelper(matcher, platformPackager.info)
     const files = await copier.collectNodeModules(dep, nodeModuleExcludedExts)
     result[index++] = validateFileSet({ src: source, destination, files, metadata: copier.metadata })
 
     if (dep.conflictDependency) {
       for (const c of dep.conflictDependency) {
-        await collectNodeModules(c, path.join(destination, NODE_MODULES, c.name))
+        await collectNodeModules(c, realSource, path.join(destination, NODE_MODULES, c.name))
       }
     }
   }
 
   for (const dep of deps) {
     const destination = path.join(mainMatcher.to, NODE_MODULES, dep.name)
-    await collectNodeModules(dep, destination)
+    const realSource = getRealSource(dep.name, dep.dir)
+    await collectNodeModules(dep, realSource, destination)
```

Comment:
Could you add a unit test for the issue this PR is fixing for? I'd like to start tracking the issues related to node module collector in unit tests if possible

Reply:
https://github.com/electron-userland/electron-builder/blob/d2e59f1bbdf3f7b270f23d8e1bafe805d3ec1c59/test/src/globTest.ts#L171

added.

Reply:
Without this fix, adding this UT (Unit Test) case directly would fail.
<img width="1063" alt="image" src="https://github.com/user-attachments/assets/42700391-184e-4cd9-ba4a-859faa81aef0">